File: commands.py
from okx_api import get_crypto_price
from subscriptions import subscriptions
import schedule
from wechaty import Contact

async def handle_subscription(msg, crypto, threshold):
    user_id = msg.talker().contact_id
    if user_id not in subscriptions:
        subscriptions[user_id] = []
    subscriptions[user_id].append((crypto, threshold))
    await msg.say(f"订阅成功: 当 {crypto.upper()} 价格达到 {threshold} USDT 时，将通知你。")

async def send_report(msg, crypto, interval):
    user_id = msg.talker().contact_id
    def report_job():
        price_info = get_crypto_price(crypto)
        user = await Contact.find(user_id)
        await user.say(f"{crypto.upper()} 定时报告:\n{price_info}")

    schedule.every(interval).minutes.do(report_job)
    await msg.say(f"订阅成功: 每 {interval} 分钟发送一次 {crypto.upper()} 的价格报告。")

async def handle_command(msg):
    content = msg.text()
    if content.startswith("/subscribe "):  # Example command: /subscribe BTC 30000
        _, crypto, threshold = content.split()
        await handle_subscription(msg, crypto, float(threshold))
    elif content.startswith("/report "):  # Example command: /report BTC 60
        _, crypto, interval = content.split()
        await send_report(msg, crypto, int(interval))
    elif content.startswith("/"):
        crypto = content[1:]
        price_info = get_crypto_price(crypto)
        await msg.say(price_info)

File: commands.js
const { get_crypto_price } = require('./okx_api');
const subscriptions = require('./subscriptions');

async function handle_subscription(msg, crypto, threshold) {
  const user_id = msg.talker().contact_id;
  if (!subscriptions[user_id]) {
    subscriptions[user_id] = [];
  }
  subscriptions[user_id].push({ crypto, threshold });
  await msg.say(`订阅成功: 当 ${crypto.toUpperCase()} 价格达到 ${threshold} USDT 时，将通知你。`);
}

async function send_report(msg, crypto, interval) {
  const user_id = msg.talker().contact_id;

  const report_job = async () => {
    const price_info = await get_crypto_price(crypto);
    const user = await bot.Contact.find({ id: user_id });
    await user.say(`${crypto.toUpperCase()} 定时报告:\n${price_info}`);
  };

  setInterval(report_job, interval * 60 * 1000);
  await msg.say(`订阅成功: 每 ${interval} 分钟发送一次 ${crypto.toUpperCase()} 的价格报告。`);
}

async function handle_command(msg) {
  const content = msg.text();
  if (content.startsWith("/subscribe ")) { // Example command: /subscribe BTC 30000
    const [_, crypto, threshold] = content.split();
    await handle_subscription(msg, crypto, parseFloat(threshold));
  } else if (content.startsWith("/report ")) { // Example command: /report BTC 60
    const [_, crypto, interval] = content.split();
    await send_report(msg, crypto, parseInt(interval));
  } else if (content.startsWith("/")) {
    const crypto = content.slice(1);
    const price_info = await get_crypto_price(crypto);
    await msg.say(price_info);
  }
}

module.exports = { handle_command };
